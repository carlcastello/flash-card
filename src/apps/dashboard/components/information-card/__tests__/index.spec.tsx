import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json'

import { InformationCard }  from '../index';
import { IconButton, Typography } from '@material-ui/core';


describe("Infromation Card", () => {

  const id: string = "information-card"

  let wrapper: any;
  let instance: any;
  let onEditFunction: any;
  let onDeleteFunction: any;
  let onClickFunction: any;

  beforeEach(() => {
    onEditFunction = jest.fn();
    onDeleteFunction = jest.fn();
    onClickFunction = jest.fn();

    wrapper = shallow(
      <InformationCard
        id={id}
        title="Information Card"
        description="This is an Information Card"
        onEdit={onEditFunction}
        onDelete={onDeleteFunction}
        onClick={onClickFunction}
        classes={{
          inlineTypography: "inline-typography",
          paperContainer: "paper-container",
          iconButton: "icon-button",
          iconButtonContainer: "icon-button-container",
        }}
      />
    );
    instance = wrapper.instance();
  });

  describe("The Information Card", () => {
    it("Shows the Card without Subtitle", () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Typography)).toHaveLength(2);
    });
    
    it("Shows the Card with Subtitle", () => {
      wrapper.setProps({subtitle: "hello world"});
      wrapper.update();

      expect(toJSON(wrapper)).toMatchSnapshot();
      expect(wrapper.find(Typography)).toHaveLength(3);
    });
  });

  describe("State after onMouseHoverAction", () => {
    
    it("Changes the state if hasHoverEffect", () => {
      expect(wrapper.state().paperElevation).toEqual(3);
      wrapper.setProps({hasHoverEffect: false});
      wrapper.update();
      instance.onMouseHoverAction(6)();
      expect(wrapper.state().paperElevation).toEqual(3);
    });

    it("Changes the state if hasHoverEffect", () => {
      expect(wrapper.state().paperElevation).toEqual(3);
      wrapper.setProps({hasHoverEffect: true});
      wrapper.update();
      instance.onMouseHoverAction(6)();
      expect(wrapper.state().paperElevation).toEqual(6);
    });
  });

  describe("Prop Functions", () => {
    it("Calls the Edit Function when Edit Icon Click", () => {
      wrapper.find(IconButton).at(0).simulate("click", { stopPropagation() {} })
      expect(onEditFunction).toHaveBeenCalledWith(id);
    });

    it("Calls the Delete Function when Delete Icon Click", () => {
      wrapper.find(IconButton).at(1).simulate("click", { stopPropagation() {} })
      expect(onDeleteFunction).toHaveBeenCalledWith(id);
    });
  })
});