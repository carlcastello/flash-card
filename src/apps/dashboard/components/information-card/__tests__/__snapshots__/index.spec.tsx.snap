// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Infromation Card The Information Card Shows the Card with Subtitle 1`] = `
<WithStyles(ForwardRef(Paper))
  className="paper-container"
  elevation={3}
  onClick={[MockFunction]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <Styled(MuiBox)
    p={2}
  >
    <Styled(MuiBox)
      mb={1}
    >
      <WithStyles(ForwardRef(Typography))
        className="inline-typography"
        variant="h5"
      >
        Information Card
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        className="inline-typography"
        variant="subtitle1"
      >
        Â (
        hello world
        )
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      className="icon-button-container"
    >
      <WithStyles(ForwardRef(IconButton))
        className="icon-button"
        onClick={[Function]}
      >
        <EditIcon
          fontSize="inherit"
        />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(IconButton))
        className="icon-button"
        onClick={[Function]}
      >
        <DeleteIcon
          fontSize="inherit"
        />
      </WithStyles(ForwardRef(IconButton))>
    </Styled(MuiBox)>
    <WithStyles(ForwardRef(Typography))
      variant="body2"
    >
      This is an Information Card
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Paper))>
`;

exports[`Infromation Card The Information Card Shows the Card without Subtitle 1`] = `
<WithStyles(ForwardRef(Paper))
  className="paper-container"
  elevation={3}
  onClick={[MockFunction]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <Styled(MuiBox)
    p={2}
  >
    <Styled(MuiBox)
      mb={1}
    >
      <WithStyles(ForwardRef(Typography))
        className="inline-typography"
        variant="h5"
      >
        Information Card
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
    <Styled(MuiBox)
      className="icon-button-container"
    >
      <WithStyles(ForwardRef(IconButton))
        className="icon-button"
        onClick={[Function]}
      >
        <EditIcon
          fontSize="inherit"
        />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(IconButton))
        className="icon-button"
        onClick={[Function]}
      >
        <DeleteIcon
          fontSize="inherit"
        />
      </WithStyles(ForwardRef(IconButton))>
    </Styled(MuiBox)>
    <WithStyles(ForwardRef(Typography))
      variant="body2"
    >
      This is an Information Card
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Paper))>
`;
